web-common/src/features/dashboards/state-managers/selectors/measures.ts

```diff
@@ -51,11 +52,27 @@ export const isMeasureValidPercentOfTotal = ({
   };
 };
 
-export const filterBasicMeasures = (
-  measures: MetricsViewSpecMeasureV2[] | undefined,
-) => measures?.filter((m) => !m.window) ?? [];
+export const filteredBasicMeasures = ({
+  metricsSpecQueryResult,
+  dashboard,
+}: DashboardDataSources) => {
+  return (includeComparisonMeasure: boolean) => {
+    includeComparisonMeasure =
+      includeComparisonMeasure &&
+      !!dashboard.showTimeComparison &&
+      !!dashboard.selectedComparisonTimeRange;
+    return (
+      metricsSpecQueryResult.data?.measures?.filter(
+        (m) =>
+          !m.window &&
+          (includeComparisonMeasure ||
+            m.type !== MetricsViewSpecMeasureType.MEASURE_TYPE_TIME_COMPARISON),
+      ) ?? []
+    );
+  };
+};
 
-export const getMeasuresAndDimensions = ({
+export const getFilteredMeasuresAndDimensions = ({
```

Comment:
Can you add a comment at the top of this function to explain what the function filters for? One has to read the function implementation to understand what's going on, and it's quite complicated.